control-plane-operator/controllers/hostedcontrolplane/pkioperator/reconcile.go

```diff
@@ -0,0 +1,184 @@
+	hyperutil "github.com/openshift/hypershift/support/util"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sutilspointer "k8s.io/utils/pointer"
+)
+
+func ReconcileServiceAccount(sa *corev1.ServiceAccount, ownerRef config.OwnerRef) error {
+	ownerRef.ApplyTo(sa)
+	return nil
+}
+
+func ReconcileDeployment(
+	deployment *appsv1.Deployment,
+	openShiftTrustedCABundleConfigMapExists bool,
+	hcp *hyperv1.HostedControlPlane,
+	cpoPKIImage string,
+	setDefaultSecurityContext bool,
+	sa *corev1.ServiceAccount,
+	certRotationScale time.Duration,
+) error {
+	deployment.Spec = appsv1.DeploymentSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"name": "control-plane-pki-operator",
+			},
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"name":                        "control-plane-pki-operator",
+					"app":                         "control-plane-pki-operator",
+					hyperv1.ControlPlaneComponent: "control-plane-pki-operator",
+				},
+			},
+			Spec: corev1.PodSpec{
+				ImagePullSecrets: []corev1.LocalObjectReference{
+					{
+						Name: "pull-secret",
+					},
+				},
+				ServiceAccountName: sa.Name,
+				Containers: []corev1.Container{
+					{
+						Name:            "control-plane-pki-operator",
+						Image:           cpoPKIImage,
+						ImagePullPolicy: corev1.PullIfNotPresent,
+						Env: []corev1.EnvVar{
+							{
+								Name: "POD_NAME",
+								ValueFrom: &corev1.EnvVarSource{
+									FieldRef: &corev1.ObjectFieldSelector{
+										FieldPath: "metadata.name",
+									},
+								},
+							},
+							{
+								Name: "HOSTED_CONTROL_PLANE_NAMESPACE",
+								ValueFrom: &corev1.EnvVarSource{
+									FieldRef: &corev1.ObjectFieldSelector{
+										FieldPath: "metadata.namespace",
+									},
+								},
+							},
+							{
+								Name:  "HOSTED_CONTROL_PLANE_NAME",
+								Value: hcp.Name,
+							},
+							{
+								Name:  "CERT_ROTATION_SCALE",
+								Value: certRotationScale.String(),
+							},
+						},
+						Command: []string{"/usr/bin/control-plane-pki-operator"},
+						Args:    []string{"operator"},
+						Ports:   []corev1.ContainerPort{{Name: "metrics", Protocol: "TCP", ContainerPort: 8443}},
+					},
+				},
+			},
+		},
+	}
+
+	if openShiftTrustedCABundleConfigMapExists {
+		hyperutil.DeploymentAddOpenShiftTrustedCABundleConfigMap(deployment)
+	}
+
+	mainContainer := hyperutil.FindContainer("control-plane-pki-operator", deployment.Spec.Template.Spec.Containers)
+	proxy.SetEnvVars(&mainContainer.Env)
+
+	deploymentConfig := config.DeploymentConfig{
+		Scheduling: config.Scheduling{
+			PriorityClass: config.DefaultPriorityClass,
+		},
+		SetDefaultSecurityContext: setDefaultSecurityContext,
+		AdditionalLabels: map[string]string{
+			config.NeedManagementKASAccessLabel: "true",
```

Comment:
to pass e2e you'll need to add this component to the list of allowed ones in EnsureNetworkPolicies

Reply:
I already did that and pass e2e? https://github.com/openshift/hypershift/pull/3193/commits/13094ab04b07808814f26cc6a3ca2c06d39ae43c#diff-cebde6b83b4b372162b2c6a22710a4ab0b1838576120535593a66a3a9bdfc170R767